### PR TITLE
Load combined file with bogus key path

### PR DESCRIPTION
**Pull request by shinrich**
Fixes an issue we see in production with 9.0.x.  We have both the cert and the key in the same file.  Our proxy.config.ssl.server.private_key.path is set to a default value different from the value of proxy.config.ssl.client.cert.path.  But the original logic, would look for the cert pem file in the directory specified in proxy.config.ssl.server.private_key.path.  This would fail, the key would not be loaded and no TLS traffic would pass.

Added an autest to exercise this issue.

### CODE REVIEW

**Comment by shinrich** [comment]
[approve ci autest]

File: tests/gold_tests/tls/tls_check_dual_cert_selection2.test.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    'ssl_cert_name=combined-ec.pem,combined.pem',
+    'ssl_cert_name=signed-foo-ec.pem,signed-foo.pem',
+    'dest_ip=* ssl_cert_name=signed-san-ec.pem,signed-san.pem'
+])
+
+# Case 1, global config policy=permissive properties=signature
+#         override for foo.com policy=enforced properties=all
+ts.Disk.records_config.update({
+    'proxy.config.ssl.server.cert.path': '{0}'.format(ts.Variables.SSLDir),
+    'proxy.config.ssl.server.key.path': '/tmp', # Faulty key path should not matter, since there are no key files
```
Comment (maskit) [inline tests/gold_tests/tls/tls_check_dual_cert_selection2.test.py:63]: proxy.config.ssl.server.private_key.path ?

Comment (shinrich) [reply tests/gold_tests/tls/tls_check_dual_cert_selection2.test.py:63]: So it is.  Odd though the test is looking for keys in /tmp.  Will dig in.

**Comment by shinrich** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**